TripCheckins/Common/Observable.swift

```diff
@@ -0,0 +1,18 @@
+//
+//  Observable.swift
+//  TripCheckins
+//
+//  Created by Nataliya  on 9/12/17.
+//  Copyright © 2017 Nataliya Patsovska. All rights reserved.
+//
+
+import Foundation
+
+protocol Observable {
```

Comment:
it might be interesting to think of default implementations here.

Reply:
Hm yes... But any ideas without exposing a mutable `observers` property in the protocol? Maybe by defining my own `Observers` collection that can fix the weak references problem and define the add/remove methods?